riemann/tests/tx/test_tx.py

```diff
@@ -855,6 +854,59 @@ def test_sighash_forkid_all_anyone_can_pay(self):
             sighash,
             helpers.SIGHASH_FORKID['all_anyone_can_pay'])
 
+    def test_hash_outputs(self):
+        t = tx.Tx(self.version, self.none_flag, self.tx_ins,
+                  self.tx_outs, self.none_witnesses, self.lock_time)
+        output = t._hash_outputs(index=0, sighash_type=tx.SIGHASH_SINGLE)
+        expected = utils.hash256(self.tx_outs[0].to_bytes())
+        self.assertEqual(output, expected)
+
+        with self.assertRaises(NotImplementedError) as context:
+            t = tx.Tx(self.version, self.none_flag, self.tx_ins,
+                      self.tx_outs, self.none_witnesses, self.lock_time)
+            t._hash_outputs(index=3, sighash_type=tx.SIGHASH_SINGLE)
+
+        self.assertIn(
+            'I refuse to implement the SIGHASH_SINGLE bug.',
+            str(context.exception))
+
+    def test_script_code(self):
+        self.stack_script = helpers.P2SH['ser']['ins'][0]['stack_script']
+        self.redeem_script = helpers.P2SH['ser']['ins'][0]['redeem_script']
+        self.sequence = helpers.P2SH['ser']['ins'][0]['sequence']
+        self.outpoint = helpers.P2SH['ser']['ins'][0]['outpoint']
+
+        self.tx_in = tx.TxIn(self.outpoint, self.stack_script,
+                             self.redeem_script, self.sequence)
+        self.tx_ins = [self.tx_in]
+
+        t = tx.Tx(self.version, self.none_flag, self.tx_ins, self.tx_outs,
+                  self.none_witnesses, self.lock_time)
+
+        bob = t.script_code(index=0)
```

Comment:
Rename variable to something clear.
Or just skip naming the variable, and include this in the asserEqual on the next line